### PR TITLE
Adding post_push hook to auto-build

### PR DESCRIPTION
**Pull request by jerryyin**
This script will be triggered when an auto-build finishes. Please refer to [here ](https://docs.docker.com/docker-hub/builds/advanced/)for definition of auto-builds.

Environmental variables used in the script:

- `DOCKER_REPO`: the name of the Docker repository being built.
- `DOCKER_TAG`: the Docker repository tag being built.
- `IMAGE_NAME`: the name and tag of the Docker repository being built. (This variable is a combination of DOCKER_REPO:DOCKER_TAG.)

Say that the auto-build creates a image called `rocm/tensorflow-autobuilds:latest`, this script will tag the image to a different name `rocm/tensorflow-autobuilds:"YYMMDD"`, eg: `rocm/tensorflow-autobuilds:190522`

Test done:
Everything are already tested in my personal fork of [tensorflow-upstream](https://github.com/jerryyin/tensorflow-upstream/tree/develop-upstream/tensorflow/tools/ci_build/hooks).

-----------

Immediately after this PR I will create another hook called `pre_build`, such that it will only build when there are code changes to `ci_build` folder

### CODE REVIEW

**Comment by jerryyin** [comment]
> The overall concept is a great idea. Might consider the following:
> 
> * Add another Dockerfile for building TF itself.  Not sure where the best location would be.  But it would use a `FROM` with the docker image that was created by `ci_build/Dockerfile.rocm`
> * Add a more detailed image tag (beyond `%y%m%d`), especially if the plan is to push an image for `develop-upstream`-related events.  Would need to be descriptive enough to allow us to tie together the image tag with the event (e.g. the commit, PR, or ..).

Nice feedback. I've thought of both when working with it in the past week.
- I agree this should be the direction to go. One question is that currently the Dockerhub repository is private, I'm not sure if we want to push the repository/tag to a public space. (In the mean time, I have created a private github repo that has this Dockerfile and also attached autobuilds to it. I can share the Dockerfile, so that everyone can use it as a template for their own autobuilds, to add their most frequently used tool, etc...)
- Adding commit id or PR comment is a one line change. One potential question is that if we updated the Dockerfile a few times a day, it will create multiple images for one day. I'm not sure if we want to go to that granularity. Having it tagged by date only implicitly keep the final commit of the date, should there be more images otherwise.

To sum up, I don't feel a strong opinion to both points. I can update now if you feel either is important. Or we can start with what we have now and update anytime in next PR.

**Comment by parallelo** [comment]
Yeah, I don't have a strong opinion on these either.  Was trying to think about what would make it most useful in a general case.  

Let's start with incremental steps.  LGTM

**Comment by sunway513** [comment]
> * One question is that currently the Dockerhub repository is private, I'm not sure if we want to push the repository/tag to a public space.

To me it's fine to public the rocm/tensorflow-autobuilds dockerhub repo, that can be a good reference for our community users as well. 
Just I'd like to keep the public rocm/tensorflow dockerhub repo clean, and have it only contains well-tested docker containers.

**Comment by jerryyin** [comment]
@sunway513 I can setup a separate automated builds in the public repository. Maybe name it as `rocm/tensorflow-nightly`? Since all our commits have already passed CI checks, I think those images qualify for being stable ones.

Regarding path and filename, does `tensorflow-upstream/tensorflow/tools/ci_build/` and `Dockerfile.rocm-nightly` sounds good to you?
